gossip-protocol-spec.md

```diff
@@ -152,29 +187,124 @@ struct Bloom {
 </details>
 
 ### PullResponse
-These are sent in response to a `PullRequest`.
+These are sent in response to a `PullRequest`. They contain filtered values from the node's `crds` which the origin node is missing. 
 
 | Data | Type | Size | Description |
 |------|:----:|:----:|-------------|
 | `Pubkey` | `[u8; 32]` | 32 | a public key of the origin |
 | `CrdsValuesList` | [`[CrdsValue]`](#data-shared-between-nodes) | 8+ | a list of new values  |
 
+<details>
+  <summary>Solana client Rust implementation</summary>
 
+```rust
+enum Protocol {
+    //...
+    PullResponse(Pubkey, Vec<CrdsValue>),
+    //...
+}
+```
+
+</details>
 
-### PushMessage
-It is sent by nodes who want to share information with others. Node receiving the message checks for:
-- duplication - duplicated messages are dropped, and the node responds with a prune message if the message came from a low staked node
-- new data:
-    - new information is stored in `crds` and replaces old value
-    - stores message in `pushed_once`, which is used for detecting duplicates
-    - retransmits information to its peers
-- expiration - messages older than `PUSH_MSG_TIMEOUT` are dropped
+
+### PruneMessage
+It is sent to peers with a list of nodes that should be pruned. No more push messages from pruned nodes should be sent by the recipient of this message to the origin.
```

Comment:
this is not exactly correct:
> No more push messages from pruned nodes should be sent by the recipient of this message to the origin.

Prunes have two components
1) The peer to prune
2) The origin for which the peer should be pruned

The peer is pruned ONLY for messages created by the origin, not for all messages. The origin is the node that created the message originally.

Reply:
we're not sending to the origin. the origin was the one that created the message.

Reply:
I phrased that sentence poorly. By origin here I've meant the origin of the PruneMessage. What about this: _No more push messages from pruned origin nodes should be sent by the recipient of this prune message to its sender._

Reply:
ya that is better! could be worth defining some naming conventions like:
1) Node: a validator running gossip
2) Peer: a node sending or receiving messages to the current node we're talking about (or like subject node or something)
3) Origin: the original creator of a message
etc...

not worded super well here but could be useful.

Reply:
I've added a list of naming conventions at the beginning of the document.

Reply:
looks great! thank you!